apps/api/v2/src/modules/organizations/services/event-types/organizations-event-types.service.ts

```diff
@@ -0,0 +1,155 @@
+    body: CreateTeamEventTypeInput_2024_06_14
+  ) {
+    const eventTypeUser = await this.getUserToCreateTeamEvent(user, orgId);
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { hosts, assignAllTeamMembers, ...rest } =
+      await this.inputService.transformInputCreateTeamEventType(teamId, body);
+    const { eventType: eventTypeCreated } = await createEventType({
+      input: { teamId: teamId, ...rest },
+      ctx: {
+        user: eventTypeUser,
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        prisma: this.dbWrite.prisma,
+      },
+    });
+
+    return await this.updateTeamEventType(
+      eventTypeCreated.id,
+      teamId,
+      { hosts: body.hosts, assignAllTeamMembers },
+      user
+    );
+  }
+
+  async getUserToCreateTeamEvent(user: UserWithProfile, organizationId: number) {
+    const isOrgAdmin = await this.membershipsRepository.isUserOrganizationAdmin(user.id, organizationId);
+    const profileId = user.movedToProfileId || null;
+    return {
+      id: user.id,
+      role: user.role,
+      organizationId: user.organizationId,
+      organization: { isOrgAdmin },
+      profile: { id: profileId },
+      metadata: user.metadata,
+    };
+  }
+
+  async getTeamEventType(teamId: number, eventTypeId: number) {
+    const eventType = await this.organizationEventTypesRepository.getTeamEventType(teamId, eventTypeId);
+
+    if (!eventType) {
+      return null;
+    }
+
+    return this.outputService.getResponseTeamEventType(eventType);
+  }
+
+  async getTeamEventTypes(teamId: number) {
+    const eventTypes = await this.organizationEventTypesRepository.getTeamEventTypes(teamId);
+
+    const eventTypePromises = eventTypes.map(async (eventType) => {
+      return await this.outputService.getResponseTeamEventType(eventType);
+    });
+
+    return await Promise.all(eventTypePromises);
+  }
+
+  async getTeamsEventTypes(orgId: number, skip = 0, take = 250) {
+    const eventTypes = await this.organizationEventTypesRepository.getTeamsEventTypes(orgId, skip, take);
+
+    const eventTypePromises = eventTypes.map(async (eventType) => {
+      return await this.outputService.getResponseTeamEventType(eventType);
+    });
+
+    return await Promise.all(eventTypePromises);
+  }
+
+  async updateTeamEventType(
+    eventTypeId: number,
+    teamId: number,
+    body: UpdateTeamEventTypeInput_2024_06_14,
+    user: UserWithProfile
+  ) {
+    const eventTypeUser = await this.eventTypesService.getUserToUpdateEvent(user);
+    const bodyTransformed = await this.inputService.transformInputUpdateTeamEventType(
+      eventTypeId,
+      teamId,
+      body
+    );
+
+    await updateEventType({
+      input: { id: eventTypeId, ...bodyTransformed },
+      ctx: {
+        user: eventTypeUser,
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        prisma: this.dbWrite.prisma,
+      },
+    });
```

Comment:
is this function making sure that the eventtype belongs to the teamId inside of the bodyTransformed ?

Reply:
Yes - it fetches event type by id, and then if teamId is passed checks if event type team id is same as passed teamId.